client/gutenberg/editor/controller.js

```diff
@@ -119,23 +117,37 @@ export const post = async ( context, next ) => {
 	const postType = determinePostType( context );
 	const isDemoContent = ! postId && has( context.query, 'gutenberg-demo' );
 
-	let state = context.store.getState();
-	let siteId = getSelectedSiteId( state );
-	if ( ! siteId ) {
-		siteId = await waitForSelectedSiteId( context );
-		state = context.store.getState();
-	}
-	const siteSlug = getSelectedSiteSlug( state );
-	const userId = getCurrentUserId( state );
+	const makeEditor = siteId => {
+		const state = context.store.getState();
+		const siteSlug = getSelectedSiteSlug( state );
+		const userId = getCurrentUserId( state );
+
+		return import( './init' ).then( ( { initGutenberg } ) => {
+			//set postId on state.ui.editor.postId, so components like editor revisions can read from it
+			context.store.dispatch( { type: EDITOR_START, siteId, postId } );
+
+			const Editor = initGutenberg( userId, siteSlug );
 
-	//set postId on state.ui.editor.postId, so components like editor revisions can read from it
-	context.store.dispatch( { type: EDITOR_START, siteId, postId } );
+			return () => <Editor { ...{ siteId, postId, postType, uniqueDraftKey, isDemoContent } } />;
+		} );
+	};
 
-	const GutenbergEditor = initGutenberg( userId, siteSlug );
+	const EditorLoader = asyncLoader( {
+		promises: {
+			Editor: waitForSiteId( context.store ).then( makeEditor ),
+			translations: loadTranslations( context.store ),
```

Comment:
@ockham this is where we can add availability info:

```js
promises: {
	Editor: waitForSiteId( context.store ).then( makeEditor ),
	availability: waitForData( { a: requestFromUrl( '…' ) } ).then( ( { a } ) => a ),
	translations: loadTranslations( context.store ),
},
…
success: ( { Editor, availability } ) => <Editor availability={ availability } />
```

Reply:
To make the `availability` prop really work, `makeEditor` needs to return `props => <Editor { ...props } />` rather than `() => <Editor />`

Reply:
why is that? why couldn't we add it as shown here?

Reply:
`Editor` returned from `makeEditor` is a functional component that ignores it props.
```js
OuterEditor = () => <InnerEditor siteId />;
<OuterEditor availability />;
```
The `availability` prop won't be passed to `InnerEditor`.

Reply:
Makes sense. I'll push a fix.

Reply:
fa397c87e2